app/controllers/concerns/neighbourhood_filter.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+class NeighbourhoodFilter
```

Comment:
Could be called `PartnerNeighbourhoodFilter` as the filter is specific to partners